core/src/main/java/io/questdb/cutlass/pgwire/modern/PGPipelineEntry.java

```diff
@@ -1392,26 +1395,36 @@ private void msgExecuteSelect(
             sqlExecutionContext.getCircuitBreaker().resetTimer();
             sqlExecutionContext.setCacheHit(cacheHit);
             try {
-                RecordMetadata oldMeta = factory.getMetadata();
+                RecordMetadata oldMeta = null;
                 for (int attempt = 1; ; attempt++) {
-                    try {
-                        copyParameterValuesToBindVariableService(
-                                sqlExecutionContext,
-                                characterStore,
-                                utf8String,
-                                binarySequenceParamsPool
-                        );
-                        cursor = factory.getCursor(sqlExecutionContext);
-                        break;
-                    } catch (TableReferenceOutOfDateException e) {
-                        if (attempt == maxRecompileAttempts) {
-                            throw e;
+                    // check if factory is null, what might happen is that
+                    // prepared statement (entry we held on to) failed to compile, factory is null
```

Comment:
consider this scenario:
1. a table `tab` has a column `col1, col2`
2. a client creates a named prepared statement, describes, binds and executes: `select * from tab`. 
3. the table is dropped -> on the next execution the factory is set to `null` and an error is reported to a user.
4. table `tab` is created again. with different columns -> the query text successfully compiles again. but: 
4a. metadata validation is skipped. so the result set we sent to a client is different than what server sent as DESCRIBE response in the step 2
4b. binding variables are not used since `copyParameterValuesToBindVariableService()` was skipped.

Reply:
@jerrinot this is along the lines of what i was thinking too, but i ran out of gas to do it. I think whether we need to recompile query or error permanently should be subject to copying existing behaviour of postgres. There are pros and cons for both approaches (e.g. error permanently or recompile and run)